nixnet_examples/programmatic_database_api.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import print_function
+
+import pprint
+import six
+import time
+
+import nixnet
+from nixnet import constants
+from nixnet import types
+from nixnet.system import system
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+def write_can_frame_using_alias(alias):
+    database_name = alias.database_name
+    cluster_name = 'CAN_Cluster'
+    output_frame = 'CANEventFrame1'
+    interface = 'CAN1'
+
+    with nixnet.FrameOutQueuedSession(
+            interface,
+            database_name,
+            cluster_name,
+            output_frame) as output_session:
+        terminated_cable = six.moves.input('Are you using a terminated cable (Y or N)? ')
+        if terminated_cable.lower() == "y":
+            output_session.intf.can_term = constants.CanTerm.OFF
+        elif terminated_cable.lower() == "n":
+            output_session.intf.can_term = constants.CanTerm.ON
+        else:
+            print("Unrecognised input ({}), assuming 'n'".format(terminated_cable))
+            output_session.intf.can_term = constants.CanTerm.ON
+
+        user_value = six.moves.input('Enter payload [int, int]: ')
+        try:
+            payload_list = [int(x.strip()) for x in user_value.split(",")]
+        except ValueError:
+            payload_list = [2, 4, 8, 16]
+            print('Unrecognized input ({}). Setting data buffer to {}', user_value, payload_list)
+
+        id = types.CanIdentifier(0)
+        payload = bytearray(payload_list)
+        frame = types.CanFrame(id, constants.FrameType.CAN_DATA, payload)
+
+        print('Writing CAN frames using ({}) alias:', database_name)
+
+        i = 0
+        while i < 4:
+            for index, byte in enumerate(payload):
+                payload[index] = byte + i
+
+            frame.payload = payload
+            output_session.frames.write([frame])
+            print('Sent frame with ID %s payload: %s' % (id, payload))
+
+def main():
+    with system.System() as my_system:
+        print(list(my_system.aliases))
+
+        print('Adding three database aliases...')
+        database_alias1 = 'custom_database1'
+        database_filepath1 = '../databases/custom_database1.dbc'
+        default_baud_rate1 = '500000'
+        alias1 = Alias(database_alias1, database_filepath1, default_baud_rate1)
+        my_system.aliases[alias1.database_name] = alias1
+        
+        database_alias2 = 'custom_database2'
+        database_filepath2 = '../databases/custom_database2.dbc'
+        default_baud_rate2 = '250000'
+        alias2 = Alias(database_alias2, database_filepath2, default_baud_rate2)
+        my_system.aliases[alias2.database_name] = alias2
+        
+        database_alias3 = 'custom_database3'
+        database_filepath3 = '../databases/custom_database3.dbc'
+        default_baud_rate3 = '800000'
+        alias3 = Alias(database_alias3, database_filepath3, default_baud_rate3)
+        my_system.aliases[alias3.database_name] = alias3
+        print(list(my_system.aliases))
+
```

Comment:
Since the intent of the example is to demonstrate the use of the database functions, can we call the alias functions first to show they are being created in preparation for the session objects.